src/compiler/types.ts

```diff
@@ -1007,6 +1013,10 @@ namespace ts {
         expression: LeftHandSideExpression;
         typeArguments?: NodeArray<TypeNode>;
     }
+    export interface SupportedExpressionWithTypeArguments extends ExpressionWithTypeArguments {

```

Comment:
This could be better named. (As could `isSupportedExpressionWithTypeArguments`). Any suggestions?